homeassistant/components/husqvarna_automower/__init__.py

```diff
@@ -52,8 +73,9 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Handle unload of an entry."""
-    coordinator: AutomowerDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
-    await coordinator.shutdown()
+    automower_entry_data: AutomowerEntryData = hass.data[DOMAIN][entry.entry_id]
+    await automower_entry_data.coordinator.shutdown()
+    automower_entry_data.listen_task.cancel()
```

Comment:
We don't need to cancel here either if we create the listen task with the config entry helper for background tasks. And then we don't need to store the task either in hass.data.